catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -535,15 +552,22 @@ private ContentItem generateContentFile(ContentItem item, Path contentDirectory)
 
         long copy;
 
-        try (InputStream inputStream = item.getInputStream()) {
-            copy = Files.copy(inputStream, contentItemPath);
-        }
+        if (storeReference != null) {
+            copy = item.getSize();
+            Files.write(Paths.get(contentItemPath.toString() + "." + REF_EXT),
+                    storeReference.getBytes(Charset.forName("UTF-8")));
+            contentItemPath = Paths.get(storeReference);
+        } else {
+            try (InputStream inputStream = item.getInputStream()) {
+                copy = Files.copy(inputStream, contentItemPath);
+            }
 
-        if (copy != item.getSize()) {
-            LOGGER.info("Created content item {} size {} does not match expected size {}",
-                    item.getId(),
-                    copy,
-                    item.getSize());
+            if (copy != item.getSize()) {
+                LOGGER.warn("Created content item {} size {} does not match expected size {}",
```

Comment:
It probably means *something* is wrong? @stustison would checking filesystem/network integrity be an appropriate action here? Not sure what else would cause a size mismatch.